common/types/wire/activation_v2.go

```diff
@@ -0,0 +1,71 @@
+package wire
+
+import "github.com/spacemeshos/go-spacemesh/common/types/primitive"
+
+//go:generate scalegen
+
+type ActivationTxV2 struct {
+	InnerActivationTxV2
+
+	SmesherID [32]byte
+	Signature [64]byte
+}
+
+type InitialAtxPartsV2 struct {
+	CommitmentATX primitive.Hash32
+	InitialPost   PostV2
+	NodeID        primitive.Hash32 // needed make hash of the first InnerActivationTxV2 unique
+}
+
+type InnerActivationTxV2 struct {
+	PublishEpoch uint32
```

Comment:
As we're changing ATX structure, it would be nice if we could include an (untrusted) `LayerID` here for sync purposes

Reply:
Could you explain what the use case is and what value the `LayerID` would have? Would it need to be part of `InnerActivationTx` (for signing)?

Reply:
It is needed for [syncv2](https://github.com/spacemeshos/go-spacemesh/pull/5769) so that better clustering of ATX IDs during sync can be achieved, reducing sync traffic. The `LayerID` must be signed so it can't be changed by other nodes that process the ATX, confusing the sync algorithm by assigning the same ATX to different layers. It is assumed that some nodes may lie in that `LayerID` field in ATXs they publish, but with majority of nodes being honest, that should not cause substantial performance degradation of the set reconciliation algorithm

Reply:
Would `LayerID` field contain the layer in which the ATX was created or something else?

Reply:
I think `PublishEpoch` needs to be the first field in the type, because some decoding logic depends on it and we cannot change older ATXs retroactively to now use the `LayerID` instead.

I'm not fully understanding how `LayerID` can help improve sync? An ATX isn't published in any layer - protocol wise it should make no difference if an ATX is published on the first day or the last day of an epoch?

Also `PublishEpoch` is part of the PoET challenge. So if it would be a `LayerID` an honest node would need to know in advance with 5 minutes accuracy when it will publish its ATX 2 weeks later, which seems very difficult to predict (especially in a multi-smesher setup) as a) there is some delay when the PoEt proof will be available, it could take multiple passes to generate the PoST proof, the node might not be in sync before publishing the ATX further delaying the publication, etc.

Any sensible node will then just set the LayerID to the last layer before the next PoET round starts to give itself the maximum time to generate the PoST and publish the ATX, causing all ATXs of the same PoET phase to be published in the same layer...

Reply:
@mafa, the ATX could contain a `LayerID` which could be transformed into an `EpochID` for the `NipostChallenge`.

Reply:
Still, we need to keep the publish epoch the first field in the encoding for backwards compatibility and so we know which version of the ATX the bytes after that encode.

We would then need to additionally encode the `LayerID` which the publishing node has no incentive to be honest about and the receiving node has no way to verify if it is correct or not. What happens if someone with a lot of identities just claims they were all published in the same layer if they weren't? 

If it's not an issue for a successful sync then why is helpful to have it? If it does cause problems - how can we prevent a malicious user from disrupting the network by claiming incorrect publication layers?

Reply:
> Still, we need to keep the publish epoch the first field in the encoding for backwards compatibility and so we know which version of the ATX the bytes after that encode.

:+1: 

@ivan4th Could you motivate and explain adding the `LayerID` field?

Reply:
@poszu The [range-based set reconciliation algorithm](https://arxiv.org/pdf/2212.13567) is based on subdividing the ranges of possible key values for which the fingerprints (hashes of keys XORed together) don't match. If keys are just hash values, and say a peer has 100 hashes that the other one doesn't have, then syncing these 100 keys which are evenly spread across the key space may cause 100 subdivisions (roughly speaking), narrowing down to each of the keys individually.

But if the keys include `LayerID` and thus can be sorted by time (layer), then the most likely situation is that these 100 keys will be close to one another, and with the algorithm we use it may only take just a few subdivisions to sync all of the keys. Basically, just narrow down to the 100 new keys with just small "already synced" gaps between quickly and then transfer all of the IDs in the time-based range, downloading the missing blobs afterwards.

With such an optimization, the new sync algorithm will be very efficient at handling "send me all of the recent data" case, which is expected to be most common for the synced / mostly synced nodes

Reply:
Thanks for the explanation. I don't see any downside to using a publish layer instead of a publish epoch. The ATX handler could translate it to the epoch easily.

Reply:
I do understand your goal but I still fail to see the benefit.

Any honest node will optimize their post proofing times to consume most (if not all) of their CG, so most of the ATXs will be published towards the end of a CG, causing all ATXs to be concentrated in very few layers.

Even if 100% of all nodes are honest, the great majority of ATXs will be within those few layers (the last layers of every CG and the last few layers of the epoch). Which makes this optimization much less efficient as it would be if ATXs would be spread evenly across an epoch.